test/unit/org/apache/cassandra/utils/binlog/BinLogTest.java

```diff
@@ -160,16 +156,16 @@ protected String type()
 
             public void writeMarshallablePayload(WireOut wire)
             {
-
+                ready.countDown();
             }
 
             public void release()
             {
                 releaseCount.incrementAndGet();
             }
         });
-        Thread.sleep(1000);
-        assertEquals(2, releaseCount.get());
+        ready.await();
```

Comment:
Did you consider using the timeout here to avoid a long wait in case if the BinLog records are not being consumed for unknown reasons?

Reply:
yup. It makes sense. I will add a timeout and let the following spin assert to throw, if something weird happens.